File: c/chuong_01/bai006.cpp
#include <bits/stdc++.h>
using namespace std;

#define yes printf("YES\n")
#define no printf("NO\n")

int main() {
    int n; cin >> n;
    (n%2 == 0) ? yes : no;
    (n%3 == 0 && n%5 == 0) ? yes : no;
    (n%3 == 0 && n%7 != 0) ? yes : no;
    (n%3 == 0 || n%7 == 0) ? yes : no;
    (n > 30 && n < 50) ? yes : no;
    (n >= 30 && (n%2 == 0 || n%3 == 0 || n%5 == 0)) ? yes : no;
    int m = n%10;
    (n >= 10 && n <= 99 && (m == 2 || m == 3 || m == 5 || m == 7)) ? yes : no;
    (n <= 100 && n%23 == 0) ? yes : no;
    (n < 10 || n > 20) ? yes : no;
    (m%3 == 0) ? yes : no;
    return 0;
}

File: python/chuong_01/bai006.py
def main():
    n = int(input())
    yes() if n%2 == 0 else no()
    yes() if n%3 == 0 and n%5 == 0 else no()
    yes() if n%3 == 0 and n%7 != 0 else no()
    yes() if n%3 == 0 or n%7 == 0 else no()
    yes() if n > 30 and n < 50 else no()
    yes() if n >= 30 and (n%2 == 0 or n%3 == 0 or n%5 == 0) else no()
    m = n%10
    yes() if n >= 10 and n <= 99 and m in {2, 3, 5, 7} else no()
    yes() if n <= 100 and n%23 == 0 else no()
    yes() if n < 10 or n > 20 else no()
    yes() if m%3 == 0 else no()

def yes():
    print("YES")

def no():
    print("NO")

main()
